python/tank/util/shotgun.py

```diff
@@ -1434,8 +1441,12 @@ def _create_published_file(tk, context, path, name, version_number, task, commen
     # now call out to hook just before publishing
     data = tk.execute_core_hook(constants.TANK_PUBLISH_HOOK_NAME, shotgun_data=data, context=context)
 
-    log.debug("Registering publish in Shotgun: %s" % pprint.pformat(data))
-    return tk.shotgun.create(published_file_entity_type, data)
+    if dry_run:
+        log.debug("Dry run. Simply returning the data that would be sent to SG: %s" % pprint.pformat(data))
+        return data
```

Comment:
here, i think we should include the "type" key in the dictionary to make it as similar to the real thing as possible. It is also sometimes useful to know (tankpublishedfile vs publishedfile)